File: Data Structures & Algorithms/sort-colors/submission-0.java
class Solution {
    public void sortColors(int[] nums) {

        int[] buckets = new int[3];

        for(int n : nums){
            buckets[n]++;
        }
        //1 2 1
        int[] copy = new int[nums.length];

        int i = 0;
        for(int r = 0; r < buckets[0]; r++){
            nums[i] = 0;
            i++;
        }
        for(int w = 0; w < buckets[1]; w++){
            nums[i] = 1;
            i++;
        }
        for(int b = 0; b < buckets[2]; b++){
            nums[i] = 2;
            i++;
        }
        
    }
}

File: Data Structures & Algorithms/subsets-ii/submission-0.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        res = []
        curSet = []
        nums.sort()
        def dfs(i):
            nonlocal curSet, res
            if i >= len(nums):
                res.append(curSet[:])
                return
            curSet.append(nums[i])
            dfs(i+1)
            curSet.pop()
            i += 1
            while i < len(nums) and nums[i] == nums[i-1]:
                i += 1
            dfs(i)
        
        dfs(0)

        return res
            